BE/index.test.js

```diff
@@ -1,6 +1,257 @@
+  calculateAchievementsStatus,
+} from "./index";
 
-    test('adds 1 + 2 to equal 3', () => {
-        expect(1+2).toBe(3)
-    })
+import { CHALLENGE_DURATION, NUMBER_OF_ACHIEVEMENTS } from "./utils/constants";
+import { StatusState } from "./types";
 
+const dayOfChallenge = 4;
+//find the way how mock Date constructor that should return now day as today var
+// const today = Date.now();
+// let challengeDay = new Date();
+// challengeDay.setDate(new Date().getDate() - dayOfChallenge);
+// const challengeDate = Date.parse(challengeDay);
 
+const listOfChallenges = {
+  1: {
+    id: 1,
+    startDate: 1606158555000,
+    taskStatus: "Pending",
+    tasksOrder: ["2", "3", "1", "4"],
+    tasksStatus: {
+      2: {
+        state: "Pending",
+        updated: 1605559641866,
+      },
+      3: {
+        state: "Pending",
+        updated: 1605559641866,
+      },
+      1: {
+        state: "Pending",
+        updated: 1605559641866,
+      },
+      4: {
+        state: "Pending",
+        updated: 1605559641866,
+      },
+    },
+    achievementsStatus: {
+      1: {
+        state: "Success",
+        updated: 1605708892710,
+      },
+      2: {
+        state: "Pending",
+        updated: 1605708892710,
+      },
+      3: {
+        state: "Failure",
+        updated: 1605708892710,
+      },
+      10: {
+        state: "Success",
+        updated: 1605708892710,
+      },
+      11: {
+        state: "Pending",
+        updated: 1605708892710,
+      },
+    },
+  },
+  2: {
+    id: 2,
+    startDate: 1606158555000,
+    tasksStatus: "Pending",
+    tasksOrder: ["1", "3", "2", "4"],
+  },
+};
+
+const tasks = {
+  1: {
+    id: 1,
+    description: "Go to bed before 11:00 PM",
+  },
+  2: {
+    id: 2,
+    description: "Take a picture of a sunset",
+  },
+  3: {
+    id: 3,
+    description: "Make a small present for your friend without a reason",
+  },
+  4: {
+    id: 4,
+    description: "Try a new cooking recipe",
+  },
+};
+
+const achievements = {
+  1: {
+    id: 1,
+    description: "Complete each task 7 days in a row",
+    image: "#",
+  },
+  2: {
+    id: 2,
+    description: "Complete five tasks before 8:00 AM",
+    image: "#",
+  },
+  3: {
+    id: 3,
+    description: "Complete 4 Monday's tasks",
+    image: "#",
+  },
+  4: {
+    id: 4,
+    description: "Complete each task 5 days in a row",
+    image: "#",
+  },
+  5: {
+    id: 5,
+    description: "Complete 3 Monday's tasks",
+    image: "#",
+  },
+};
+
+const defaultAchievements = {
+  10: {
+    id: 10,
+    description: "Complete half of the tasks",
+    image: "#",
+  },
+  11: {
+    id: 11,
+    description: "Complete all tasks",
+    image: "#",
+  },
+};
+
+jest.mock("./data/tasks.json", () => tasks);
+jest.mock("./data/listOfChallenges.json", () => listOfChallenges);
+jest.mock("./data/achievements.json", () => achievements);
+jest.mock("./data/defaultAchievements.json", () => defaultAchievements);
+
+describe("getCurrentTask", () => {
```

Comment:
This test fails because of Date
it's better to extract days calculation in separate function and mock it returns or spy on Date implementation in beforeEach/afterEach